### DISCUSSION TITLE
Staticly link dependices into engine, editor and plugins.

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [WohlSoft/PGE-Project#598] [Q&A]

**Comment by Wohlstand** [comment]
To make a static build of the Moondust Devkit, you are required to build the static build of the Qt toolkit.

There are two articles could help you on how to build the thing, for every platform there are different ways:
- https://wohlsoft.ru/pgewiki/Building_static_Qt_5 - general explanation and guiding including installing of required dependencies
- https://wohlsoft.ru/pgewiki/Qt_5_static_build_scripts - a set of quick scripts for different cases and versions of Qt that I use by myself to produce Qt builds for macOS and Linux which are used in the mainstream. It's possible to also produce the thing for Windows, using of different command line arguments (because of different components are available for Windows). But, static builds of Qt for Windows aren't produced by me for mainstream packages because it significantly increases the total size of released packages and doesn't have sense because DevKit includes MANY components that use Qt together. On Linux, there are own Qt builds are produces to increase the compatibility between different Linux distros using the same binaries. On macOS, every application gets packed into directory tree, and typically, deployed application includes all the shared libraries into the same bundle. However, when linking the same application to the static Qt, the size of bundle is way smaller than deployment of shared-built Qt.

**Comment by Wohlstand** [comment]
Also: some of Moondust components were already built as static components on all platforms even on Windows:
- Console tools like GIFs2PNG, PNG2GIFs, LazyFixToo, and PalleteFilter: they are way more useful if they built as fully-static things. Additionally, they don't use Qt at all.
- Moondust Engine itself: it's supposed to be used in packages of released games when it will be ready for production. And so, it doesn't uses Qt at all (in the past, it used Qt, but soon, I removed Qt dependency from Engine). So, Moondust Engine as a "Runtime counterpart" (opposed to Devkit), is always a static build on all platforms.

**Comment by Shadowblitz16** [comment]
oh ok

**Comment by Wohlstand** [comment]
btw. This also going to be a discussion.
